crates/oxc_parser/src/lexer/mod.rs

```diff
@@ -1478,3 +1289,526 @@ enum SurrogatePair {
+        if lexer.next_eq('=') {
+            Kind::Eq3
+        } else {
+            Kind::Eq2
+        }
+    } else if lexer.next_eq('>') {
+        Kind::Arrow
+    } else {
+        Kind::Eq
+    }
+};
+
+// >
+const GTR: ByteHandler = |lexer| {
+    lexer.consume_char();
+    // `>=` is re-lexed with [Lexer::next_jsx_child]
+    Kind::RAngle
+};
+
+// ?
+const QST: ByteHandler = |lexer| {
+    lexer.consume_char();
+    if lexer.next_eq('?') {
+        if lexer.next_eq('=') {
+            Kind::Question2Eq
+        } else {
+            Kind::Question2
+        }
+    } else if lexer.peek() == '.' {
+        // parse `?.1` as `?` `.1`
+        if lexer.peek2().is_ascii_digit() {
+            Kind::Question
+        } else {
+            lexer.current.chars.next();
+            Kind::QuestionDot
+        }
+    } else {
+        Kind::Question
+    }
+};
+
+// @
+const AT_: ByteHandler = |lexer| {
+    lexer.consume_char();
+    Kind::At
+};
+
+// [
+const BTO: ByteHandler = |lexer| {
+    lexer.consume_char();
+    Kind::LBrack
+};
+
+// \
+const ESC: ByteHandler = |lexer| {
+    let mut builder = AutoCow::new(lexer);
+    let c = lexer.consume_char();
+    builder.push_matching(c);
+    builder.force_allocation_without_current_ascii_char(lexer);
+    lexer.identifier_unicode_escape_sequence(&mut builder, true);
+    let text = lexer.identifier_name(builder);
+    Kind::match_keyword(text)
+};
+
+// ]
+const BTC: ByteHandler = |lexer| {
+    lexer.consume_char();
+    Kind::RBrack
+};
+
+// ^
+const CRT: ByteHandler = |lexer| {
+    lexer.consume_char();
+    if lexer.next_eq('=') {
+        Kind::CaretEq
+    } else {
+        Kind::Caret
+    }
+};
+
+// `
+const TPL: ByteHandler = |lexer| {
+    lexer.consume_char();
+    lexer.read_template_literal(Kind::TemplateHead, Kind::NoSubstitutionTemplate)
+};
+
+// {
+const BEO: ByteHandler = |lexer| {
+    lexer.consume_char();
+    Kind::LCurly
+};
+
+// |
+const PIP: ByteHandler = |lexer| {
+    lexer.consume_char();
+    if lexer.next_eq('|') {
+        if lexer.next_eq('=') {
+            Kind::Pipe2Eq
+        } else {
+            Kind::Pipe2
+        }
+    } else if lexer.next_eq('=') {
+        Kind::PipeEq
+    } else {
+        Kind::Pipe
+    }
+};
+
+// }
+const BEC: ByteHandler = |lexer| {
+    lexer.consume_char();
+    Kind::RCurly
+};
+
+// ~
+const TLD: ByteHandler = |lexer| {
+    lexer.consume_char();
+    Kind::Tilde
+};
+
+const L_A: ByteHandler = |lexer| match lexer.identifier_name_handler() {
+    "abstract" => Kind::Abstract,
+    "accessor" => Kind::Accessor,
+    "any" => Kind::Any,
+    "as" => Kind::As,
+    "assert" => Kind::Assert,
+    "asserts" => Kind::Asserts,
+    "async" => Kind::Async,
+    "await" => Kind::Await,
+    _ => Kind::Ident,
+};
+
+const L_B: ByteHandler = |lexer| match lexer.identifier_name_handler() {
+    "bigint" => Kind::BigInt,
+    "boolean" => Kind::Boolean,
+    "break" => Kind::Break,
+    _ => Kind::Ident,
+};
```

Comment:
Can't you skip the first byte?
```suggestion
const L_B: ByteHandler = |lexer| match lexer.identifier_name_handler()[1:] {
    "igint" => Kind::BigInt,
    "oolean" => Kind::Boolean,
    "reak" => Kind::Break,
    _ => Kind::Ident,
};
```